easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1-intel-2018a-Python-3.6.4.eb

```diff
@@ -0,0 +1,100 @@
+easyblock = 'PythonPackage'
+
+name = 'PyTorch'
+version = '0.3.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'http://pytorch.org/'
+description = """Tensors and Dynamic neural networks in Python with strong GPU acceleration.
+PyTorch is a deep learning framework that puts Python first."""
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+
+pytorchdir = 'pytorch-%(version)s'
+extract_cmd_pattern = 'tar -C %s/%s --strip-components=1 -xf %%s'
+
+source_urls = [
+    'https://github.com/pytorch/pytorch/archive',
+    "https://github.com/facebookincubator/gloo/archive",
+    "https://github.com/google/googletest/archive",
+    "https://github.com/nanopb/nanopb/archive",
+    "https://github.com/pybind/pybind11/archive",
+    "https://github.com/wjakob/clang-cindex-python3/archive",
+]
+# PyTorch pulls in a bunch of submodules which don't have releases.
+# We download the submodule revisions from their repos.
+sources = [
+    'v%(version)s.tar.gz',  # PyTorch
+    {
+        'download_filename': 'cb002e4eb8d167c2c60fc3bdaae4e1844e0f9353.tar.gz',
+        'filename': 'gloo-20180105.tar.gz',
+        'extract_cmd': extract_cmd_pattern % (pytorchdir, 'torch/lib/gloo'),
+    },
+    {
+        'download_filename': 'ec44c6c1675c25b9827aacd08c02433cccde7780.tar.gz',
```

Comment:
@wpoely86 @smoors @hajgato @zao This commits points to an old version of `googletest`, any specific reason for that?

Reply:
They are exactly the hashes that the submodule indicated when pulling the 0.3.0 tag. I have no idea why they use the version they do.
My workflow was to do a recursive init of submodules and mechanically copy the hashes and paths into my recipe.

Reply:
It looks like they re-organized things a bit, now you need to look at https://github.com/pytorch/pytorch/tree/v0.4.1/third_party...